eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -210,6 +210,25 @@ jobs:
     displayName: Publish Code Coverage to DevOps
     condition: eq(variables['AZ_SDK_CODE_COV'], 1)
 
+  - task: mspremier.BuildQualityChecks.QualityChecks-task.BuildQualityChecks@6
+    displayName: Check line coverage
+    inputs:
+      checkCoverage: true
+      coverageFailOption: fixed
+      coverageType: line
+      # 85% minimum line coverage
+      coverageThreshold: 85
+    condition: eq(variables['AZ_SDK_CODE_COV'], 1)
```

Comment:
Does having the condition be identical for both line and branch coverage work for both? I saw you tested branch coverage failing CI, are we confident line coverage will fail CI too?

Reply:
Line coverage will fail CI as well, I tested that in this commit: https://github.com/Azure/azure-sdk-for-c/pull/535/commits/e8f9abbfb8b6241db4aa5b57d1d871576d7873a9

The condition runs the task whenever the environment variable is set to `1` regardless of preceding steps' success or failure. If both line and branch coverage drop below expected thresholds I expect both steps to fail. 

Azure DevOps will only display one set of coverage assets for a given pipeline execution so we use the condition to only evaluate coverage on the one matrix entry that has coverage explicitly enabled.